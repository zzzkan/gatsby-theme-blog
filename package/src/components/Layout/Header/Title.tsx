import React from "react"
import { Link as GatsbyLink } from "gatsby"
import { Heading, Link } from "@chakra-ui/react"
import { useSiteMetadata } from "../../../hooks/useSiteMetadata"

const Title: React.FC = () => {
  const { title } = useSiteMetadata()
  return (
    <Heading as={"div"} size={"md"} fontWeight={"extrabold"} paddingY={3}>
      <Link as={GatsbyLink} to="/" _hover={{ textDecoration: "none" }}>
        {title}
      </Link>
    </Heading>
  )
}

export default Title
