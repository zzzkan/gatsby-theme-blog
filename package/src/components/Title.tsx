import React from "react"
import { Link as GatsbyLink } from "gatsby"
import { Heading, Link } from "@chakra-ui/react"
import { useMultilingualSentence } from "../hooks/useMultilingualSentence"

type Props = {
  readonly title: string
}

export const Title: React.FC<Props> = ({ title }) => {
  const { movePageSentence } = useMultilingualSentence()
  return (
    <Heading as={"div"} size={"md"} fontWeight={"extrabold"} paddingY={3}>
      <Link
        as={GatsbyLink}
        to="/"
        aria-label={movePageSentence("/")}
        _hover={{ textDecoration: "none" }}
      >
        {title}
      </Link>
    </Heading>
  )
}
