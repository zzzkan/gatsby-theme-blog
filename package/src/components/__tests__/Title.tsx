import React from "react"
import { render } from "@testing-library/react"
import { Title } from "../Title"

describe("Title component", () => {
  vi.mock("../../hooks/useSiteMetadata", () => {
    return {
      useSiteMetadata: vi.fn().mockImplementation(() => {
        return {
          title: "title",
        }
      }),
    }
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  afterAll(() => {
    vi.restoreAllMocks()
  })

  test("snapshot", () => {
    const { asFragment } = render(<Title />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("should render home link", () => {
    const { getByText } = render(<Title />)
    expect(getByText("title")).toHaveAttribute("href", "/")
  })
})
