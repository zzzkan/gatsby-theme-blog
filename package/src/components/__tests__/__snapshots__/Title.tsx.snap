// Vitest Snapshot v1

exports[`Title component > snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-heading css-u4py6g"
  >
    <a
      class="chakra-link css-163zzif"
      href="/"
    >
      title
    </a>
  </div>
</DocumentFragment>
`;
